### DISCUSSION TITLE
Best workflow for zstd objects within binary

### DISCUSSION THREAD

**Discussion by tgross35** [WerWolv/ImHex#1161] [Get Help] [answered]
I have a binary with some chunks that are zstd compressed, like this:

![image](https://github.com/WerWolv/ImHex/assets/13724985/5dffccad-ed5f-4a24-98db-1185122260cf)

Is there a good workflow to view the uncompressed data in a separate tab or something along those lines? Also, how do you activate a builtin pattern type for a file when it isn't autodetected?

(side note, what an amazing tool this is)

**Comment by paxcut** [comment, marked as answer]
Note that ImHex does not currently decompress any of the compressed formats supported  so it wont be able to do it for zstd either.  I am not sure what you are referring to when you mention pattern activation.  An input file can import any pattern file (File>Import>Pattern file) and when you click the green play icon under the console the pattern will be executed using the input file as the data.

**Comment by tgross35** [comment]
> Note that ImHex does not currently decompress any of the compressed formats supported so it wont be able to do it for zstd either. 

I wasn't sure if there may be some way to do that with the data processor that I was missing, but thanks for the confirmation

> I am not sure what you are referring to when you mention pattern activation. An input file can import any pattern file (File>Import>Pattern file) and when you click the green play icon under the console the pattern will be executed using the input file as the data.

That's it - I just had no clue how to get to that menu to import them. Thanks!
